opencensus/exporters/trace/ocagent/internal/ocagent_exporter.cc

```diff
@@ -0,0 +1,338 @@
+    auto link = proto_span->mutable_links()->add_link();
+    link->set_trace_id(span_link.trace_id().ToHex());
+    link->set_span_id(span_link.span_id().ToHex());
+    link->set_type(ConvertLinkType(span_link.type()));
+    PopulateAttributes(
+        span_link.attributes(),
+        proto_span->mutable_attributes()->mutable_attribute_map());
+  }
+}
+
+class Handler : public ::opencensus::trace::exporter::SpanExporter::Handler {
+ public:
+  Handler(OcAgentOptions &&opts);
+
+  void Export(const std::vector<::opencensus::trace::exporter::SpanData> &spans)
+      override;
+
+ private:
+  const OcAgentOptions opts_;
+  ::opencensus::proto::agent::common::v1::Node nodeInfo_;
+  void ConnectAgent();
+  void InitNode();
+};
+
+void ConvertSpans(
+    const std::vector<::opencensus::trace::exporter::SpanData> &spans,
+    ::opencensus::proto::agent::trace::v1::ExportTraceServiceRequest *request) {
+  for (const auto &from_span : spans) {
+    auto to_span = request->add_spans();
+
+    // 1. trace_id
+    to_span->set_trace_id(from_span.context().trace_id().ToHex());
+    // 2. span_id
+    to_span->set_span_id(from_span.context().span_id().ToHex());
+    // 3. parent_span_id
+    to_span->set_parent_span_id(from_span.parent_span_id().ToHex());
+    // 4. name
+    SetTruncatableString(from_span.name(), kDisplayNameStringLen,
+                         to_span->mutable_name());
+    // 5. The start time of the span.
+    EncodeTimestampProto(from_span.start_time(), to_span->mutable_start_time());
+    // 6. The end time of the span.
+    EncodeTimestampProto(from_span.end_time(), to_span->mutable_end_time());
+
+    // 7. Export Attributes
+    ConvertAttributes(from_span, to_span);
+
+    // 8. stack_trace
+
+    // 9. Export Time Events.
+    ConvertTimeEvents(from_span, to_span);
+
+    // 10. Export Links.
+    ConvertLinks(from_span, to_span);
+
+    // 11. The status of the span.
+    to_span->mutable_status()->set_code(
+        static_cast<int32_t>(from_span.status().CanonicalCode()));
+    to_span->mutable_status()->set_message(from_span.status().error_message());
+
+    // 12. True if the parent is on a different process.
+    to_span->mutable_same_process_as_parent_span()->set_value(
+        !from_span.has_remote_parent());
+
+    // 13. child_span_count
+
+    // 14. span kind
+
+    // 15. tracestate
+
+    // 16. resource
+    // (TODO) Add FULL support.
+  }
+}
+
+Handler::Handler(OcAgentOptions &&opts) : opts_(std::move(opts)) {
+  InitNode();
+  ConnectAgent();
+}
+
+void Handler::Export(
+    const std::vector<::opencensus::trace::exporter::SpanData> &spans) {
+  ::opencensus::proto::agent::trace::v1::ExportTraceServiceRequest request;
+  ConvertSpans(spans, &request);
+  request.mutable_node()->CopyFrom(nodeInfo_);
+
+  grpc::ClientContext context;
+  context.set_deadline(absl::ToChronoTime(absl::Now() + opts_.rpc_deadline));
+
+  auto stream = opts_.trace_service_stub->Export(&context);
+  bool ok = stream->Write(request);
```

Comment:
Please report errors to `std::cerr`, that's what the other exporters currently do.